### PR TITLE
Add the first integration test

### PR DESCRIPTION
**Pull request by mvasin**
This PR adds the first test that actually interacts with WordPress core.

I think that stubs make sense in cases like interacting with external APIs. When tests involve WordPress core, there is no instability, and such broader tests (potentially with database involved) could run a bit longer, but they also test much more, so setting them up is worth the hassle.

I tried to touch the current code as little as possible, but I think that eventually we should rewrite tests to get rid of stubs. That's why I moved phpunit out of `composer.json` to `tests/composer.json`.

After/if we agree on this, I will add tests and implement the optional enforcing of SSL response when WordPress is addressed from non-443 port (WP_FORCE_SSL environment variable, see wordplate/wordplate#196).

What I'm currently stuck with is coverage reports. The reporter [does generate analysis](https://travis-ci.org/mvasin/framework/jobs/351677877#L591), and I map `tests/vendor/wordplate/framework/src` directory to `src/` according to the [docs](https://docs.codecov.io/docs/fixing-paths), but still [codecov sees nothing](https://codecov.io/gh/mvasin/framework/branch/debug2).

### CODE REVIEW

**Comment by vinkla** [comment]
Thanks for the pull request @mvasin! I'll take a look as soon as possible.

File: tests/ContainerTest.php [hunk 1]
```diff
@@ -17,6 +17,8 @@
 use WordPlate\Application;
 use WordPlate\Container;
 
+require_once __DIR__.'/stubs.php';
```
Comment (vinkla) [inline tests/ContainerTest.php:20]: Hmm, maybe we can autoload this file instead of using require?

File: composer.json [hunk 2]
```diff
@@ -54,6 +43,9 @@
             "dev-master": "6.2-dev"
         }
     },
+    "scripts": {
+        "test": "@composer --working-dir=tests test"
```
Comment (vinkla) [inline composer.json:47]: Can we move this with PHPUnit in the `.travis.yml` file?

File: phpunit.xml.dist [hunk 3]
```diff
@@ -17,15 +17,19 @@
...
         <whitelist processUncoveredFilesFromWhitelist="true">
-            <directory suffix=".php">./src</directory>
+            <directory suffix=".php">tests/vendor/wordplate/framework/src</directory>
         </whitelist>
     </filter>
     <php>
         <env name="WP_ENV" value="testing"/>
     </php>
+    <logging>
+        <log type="coverage-clover" target="clover.xml"/>
```
Comment (vinkla) [inline phpunit.xml.dist:33]: Do we need to move this here? I would prefer if this was added to the `.travis.yml` file instead.

**Comment by vinkla** [comment]
> I tried to touch the current code as little as possible, but I think that eventually we should rewrite tests to get rid of stubs.

I agree, it would be great if we could use the built-in functions.

> That's why I moved phpunit out of `composer.json` to `tests/composer.json`.

Do we really need to keep these separate? It would be great if we could keep one `composer.json` file.

> ...but still [codecov sees nothing](https://codecov.io/gh/mvasin/framework/branch/debug2).

I wonder if Codecov can handle WordPress 🤔

**Comment by mvasin** [comment]
@vinkla For some reason I missed notifications about your response, and currently I'm busy, but I'll get back to this as soon as I can.

**Comment by vinkla** [comment]
@mvasin no worries, there is no stress 😉

**Comment by vinkla** [comment]
Thanks for the pull request @mvasin! I'm closing this due to inactivity.

**Comment by mvasin** [comment]
I've been keeping this PR in my head and it's pity I couldn't find time to finish it. If someone (including me) will take this effort further later on, have a look at https://github.com/WordPress/gutenberg/issues/12313, it could help.

**Pull request closed without merging**